cypress/integration/api/alerts/device_alerts.ts

```diff
@@ -0,0 +1,137 @@
+  });
+
+  it("Cannot create alert without access permissions", () => {
+    cy.apiAddAlert(helper,alertName,conditions,camera,null,AuthorizationError).then((response) => {checkResponse(response,AuthorizationError)});
+  });
+
+  it("Cannot create alert with invalid condition", () => { 
+    const bad_conditions = [{"tag": "any", "automaticF": true}]
+    cy.apiAddAlert(user,alertName,bad_conditions,camera,null,BadRequest).then((response) => {checkResponse(response,BadRequest)});
+  });
+
+  it("Can create alert and has no events by default", () => {
+    // create alert
+    cy.apiAddAlert(user,alertName,conditions,camera,0,200).then((response)=> {
+
+    // crete an example alert to compare against
+    cy.createExpectedAlert("emptyAlert",{"id": getCreds(alertName).id,
+        "name": "alert1",
+        "frequencySeconds": 0,
+        "conditions": [{"tag":"possum","automatic":true}],
+        "lastAlert":false,
+        "User":{"id":getCreds(user).id, "username":getTestName(getCreds(user).name), "email":getTestName(getCreds(user).name)+"@api.created.com"},
+        "Device":{"id":getCreds(camera).id, "devicename":getTestName(getCreds(camera).name)}});
+    });
+
+    //check we created an alert wth no events yet
+    cy.apiCheckAlert(user,camera,"emptyAlert");
+  });
+
+  it("Can receive an alert", () => {
+    //expected alert to compare against 
+    cy.createExpectedAlert("alert1",{"id": getCreds(alertName).id,
+    "name": "alert1",
+    "frequencySeconds": 0,
+    "conditions": [{"tag":"possum","automatic":true}],
+    "lastAlert":true,
+    "User":{"id":getCreds(user).id, "username":getTestName(getCreds(user).name), "email":getTestName(getCreds(user).name)+"@api.created.com"},
+    "Device":{"id":getCreds(camera).id, "devicename":getTestName(getCreds(camera).name)}});
+    
+    //upload a recording tagged as possum and  build an expected event using the returned recording details
+    cy.uploadRecording(camera, { tags: ["possum"] }, null, "recording1").then((response)=>{
+      cy.createExpectedEvent("event1",{"id":1,
+      "dateTime":"2021-05-19T01:39:41.376Z",
+      "createdAt":"2021-05-19T01:39:41.771Z",
+      "DeviceId":getCreds(camera).id,
+      "EventDetail": {"type":"alert", "details":{"recId":getCreds("recording1").id, "alertId":getExpectedAlert(alertName).id, "success":true, "trackId":1}},
+      "Device":{"devicename":getTestName(getCreds(camera).name)}})
+    });
+     
+    //check that an alert is present and has a 'last alerted' 
+    cy.apiCheckAlert(user,camera,"alert1");
+
+    //check expected event is received
+    cy.apiCheckEvents(user,camera,"event1");
+
+  });
+
+  it("Does not alert on non-master tags", () => {
+   //upload a recording tagged as possum
+    cy.uploadRecording(camera, { model: "different",tags: ["possum"]} ,null, "recording2" );
+
+    //check we still have just 1 event
+    cy.apiCheckAlert(user,camera,"alert1");
+    cy.apiCheckEvents(user,camera,"event1");
```

Comment:
All tests should be stand alone if possible but this one isn't because the Event is from a previous "it" branch.   If you just created a new camera for the group
eg cy.apiCreateGroupAndCameras(Penny, group, camera)
Then this wouldn't be a problem and then it would be easier to understand the test.  

Another question is why would a possum tag (on master even) trigger this alert?   You have to ferret round the whole test file to find out.   

Compare this with this test:
    it("If station changes the a new visit should be created", () => {
        const group = "stations-diff";
        const camera = "camera"
        cy.apiCreateGroupAndCameras(Penny, group, camera);
    
        const stations = [
            { name: "forest", lat: -44.0, lng: 172.7},
            { name: "waterfall", lat: -43.6, lng: 172.8}
          ];
        cy.apiUploadStations(Penny, group, stations);
        cy.uploadRecording(camera, { tags: ["possum"], lat: -44.0, lng: 172.7 });
        cy.uploadRecording(camera, { tags: ["cat"], lat: -43.6, lng: 172.8});
        cy.checkMonitoring(Penny, camera, [{station: "forest"}, {station: "waterfall"}]);
    })

The details (eg lat, long for the test matter, but they aren't shown here).